### PR TITLE
Document Manager

### PR DESCRIPTION
**Pull request by dantleech**
This PR aims to integrate the DocumentManager whilst keeping the external ContentMapper API the same.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -32,7 +32,10 @@
         "symfony/symfony": "2.6.*",
         "willdurand/hateoas-bundle": "0.3.*",
         "symfony/swiftmailer-bundle": "~2.3",
-        "pagerfanta/pagerfanta": "~1.0"
+        "pagerfanta/pagerfanta": "~1.0",
+        "sulu-io/document-manager": "dev-master",
+        "sulu/document-manager": "dev-master",
+        "dantleech/decorator-generator": "dev-master"
```
Comment (danrot) [inline composer.json]: I didn't really get the description of this library... Does it just alias class names? It doesn't provide an opportunity to further extend the new class?

File: composer.json [hunk 2]
```diff
@@ -32,7 +32,10 @@
         "symfony/symfony": "2.6.*",
         "willdurand/hateoas-bundle": "0.3.*",
         "symfony/swiftmailer-bundle": "~2.3",
-        "pagerfanta/pagerfanta": "~1.0"
+        "pagerfanta/pagerfanta": "~1.0",
+        "sulu-io/document-manager": "dev-master",
```
Comment (danrot) [inline composer.json]: That package doesn't seem to exist.

Comment (dantleech) [inline composer.json]: This is not the branch I am currently working on ..

Comment (danrot) [inline composer.json]: Shouldn't this PR be closed and the branch be deleted then? Which one is the branch which should work?

Comment (dantleech) [inline composer.json]: I will merge into this branch. The other branch is `structure_manager_2_search`.

**Comment by dantleech** [comment]
Replaced by #1090

**Pull request closed without merging**